### PR TITLE
Restaurant listings

### LINKED ISSUE #5: List restaurants and show their distance from Saddleback
As a client, I want to be able to see the list of restaurants and their distances from Saddleback so that I can check them out. 

### Tasks:
* Create a simple a `.ui` file so we can use it anywhere when needed
* Display each restaurant in the UI
* Display distance from Saddleback for each restaurant

### Tests:
* Display `0` restaurants
* Display `1` restaurants
* Display `5` restaurants

### Definition of done:
* List all the restaurants in a simple UI
* Distances from Saddleback to restaurants are shown

**Priority:** Highest

### LINKED ISSUE #6: List restaurants' menu items
As a client, I want to be able to see all the menu items for any restaurant so I can understand what each restaurant provides for me.

### Tasks:
* Create a separate `.ui` file
* Display the menu items given a restaurant

### Tests:
* Display a restaurant with `0` menu items
* Display a restaurant with `1` menu items
* Display a restaurant with `5` menu items

### Definition of done:
* When a restaurant is selected and the menu button is pressed the user will be able to see the menu items of that restaurant

**Priority:** Highest

### PR DESCRIPTION
**Pull request by ChrisRzech**
Closes #5 

### Key features
* Lists restaurant names and distance given the restaurant ID
* Simple use by instantiating it with a parent widget
* Initial size takes after the parent
* Drag & drop feature from separate lists is possible
* The main window is not resizable anymore, as discussed with @RogChav 
* Logo is now being used as icons throughout the program

### Future Improvements
* When backend classes land, a setting should be able to be toggled to automatically fill the list with the backend's model

### Notes
This story requires a lot of the backend to be finished in order to *fully* complete. A future PR is expected to finish up this story; but for the sprint's sake, this PR should be landed.

### CODE REVIEW

**Review by AAE1997 — COMMENTED** [review]
Looks good so far, do we need to worry about the red lines?

**Comment by ChrisRzech** [comment]
No that's what I changed

**Review by Roger-Chavez — APPROVED** [review]
compiled and runs fine on my computer, all tasks are complete. Approved.

**Comment by AAE1997** [comment]
Everything is running smoothly as is.

File: src/widgets/restaurantlist.cpp [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    /* List widget settings */
+    QListWidget::setStyleSheet("QListWidget { background-color: #303030; color: white; }");
+    QListWidget::resize(parent->size());
+    QListWidget::setVerticalScrollMode(QAbstractItemView::ScrollPerPixel);
+    QListWidget::setFlow(QListView::LeftToRight);
+    QListWidget::setWrapping(true);
+    QListWidget::setUniformItemSizes(true);
+    QListWidget::setDefaultDropAction(Qt::DropAction::MoveAction);
+
+    //Rebroadcasts the QListWidget's signal
```
Comment (ChrisRzech) [inline src/widgets/restaurantlist.cpp]: Incorrect comment
```suggestion
    //Converts the row to and ID
```

File: src/widgets/restaurantlist.hpp [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+class RestaurantList : public QListWidget
+{
+    Q_OBJECT
+
+public:
+    /* Constructor */
+    RestaurantList(QWidget* parent);
+
+    /* Drag and drop */
+    void setDragDropMode(QAbstractItemView::DragDropMode);
```
Comment (ChrisRzech) [inline src/widgets/restaurantlist.hpp]: `setDragDropMode` and `setDropActionMode` are redundant functions, `QListWidget` already inherits functions that do these (`setDragDropMode` and `setDefaultDropAction`).

File: src/windows/login.ui [hunk 3]
```diff
@@ -51,7 +54,7 @@
       <rect>
        <x>0</x>
        <y>0</y>
-       <width>227</width>
+       <width>236</width>
```
Comment (ChrisRzech) [inline src/windows/login.ui]: remove login size changes

File: src/windows/mainwindow.hpp [hunk 4]
```diff
@@ -23,9 +24,8 @@ private slots:
     void changeView(int);
 
 private:
-    /* Events */
-    void resizeEvent(QResizeEvent*) override;
-
     Ui::MainWindow* m_ui;
     NavBar* m_navbar;
+    RestaurantList* m_restaurantList;
+    RestaurantList* m_restaurantList2;
```
Comment (ChrisRzech) [inline src/windows/mainwindow.hpp]: Unused variable

**Review by ChrisRzech — COMMENTED** [review]
🔥 🔥 🔥 /5

**Pull request merged**